### PR TITLE
Update mycli port.

### PR DESCRIPTION
**Pull request by xeron**
* new version 1.8.1
* add python 34 and 35 variants
* add python category
* remove Id line
* add noarch

Depends on #13.

### CODE REVIEW

File: databases/mycli/Portfile [hunk 1]
```diff
@@ -1,14 +1,12 @@
-# $Id$
```
Comment (mojca) [inline databases/mycli/Portfile]: It's not absolutely necessary, but you could add the modeline on top. You'll also notice that [current version](https://github.com/xeron/macports-ports/blob/c2379d1ca9d637f4696c3f800c0e04b500f86ea8/databases/mycli/Portfile) doesn't show any syntax highlighting.

File: databases/mycli/Portfile [hunk 2]
```diff
@@ -1,14 +1,12 @@
...
-
 PortSystem          1.0
 PortGroup           python 1.0
 PortGroup           github 1.0
 
-github.setup        dbcli mycli 1.5.2 v
+github.setup        dbcli mycli 1.8.1 v
 
-python.default_version     27
+python.versions     27 34 35
```
Comment (mojca) [inline databases/mycli/Portfile]: I admit that I'm confused myself, but for ports that are not python modules themselves (that is: ports that are not `python/py-foo`) I fail to see how providing multiple `python.versions` would help exactly unless you provide variants to switch to a different python version.

Given the set of files that this port provides and in case that modules are useful for anything else than just `mycli` command line itself, you could also make this a regular python module `py-mycli` and then `py27-mycli`, `py35-mycli` could coexist. Then you could add just a tiny port `mycli` to make a symlink from `$prefix/bin/mycli` to `$prefix/Library/Frameworks/Python.framework/Versions/X.Y/bin/mycli`. That's outside of scope of this PR anyway.

Basically the only question I'm concerned about is what exactly this change solves. (I don't know myself. I don't see how to switch to version 3.5 for example and even if it was possible, it would be kind of violation of conventions.)

Comment (xeron) [inline databases/mycli/Portfile]: Yeah I think I'm also not sure how it works. Given that I don't want to change it to `py-mycli` for now what should I do with `python.versions` and `python.default_version`?

Comment (xeron) [inline databases/mycli/Portfile]: BTW I think I did it because of dependencies:
```tcl
depends_build       port:py${python.version}-setuptools
depends_lib-append  port:py${python.version}-click \
                    port:py${python.version}-crypto \
                    port:py${python.version}-pygments \
                    port:py${python.version}-prompt_toolkit \
                    port:py${python.version}-pymysql \
                    port:py${python.version}-sqlparse \
                    port:py${python.version}-configobj
```

Comment (mojca) [inline databases/mycli/Portfile]: > what should I do with `python.versions` and `python.default_version`?

Either nothing (that is: remove the patch for this line) or perhaps something similar like some other ports do. A random example:
~~~tcl
variant python26 conflicts python27 description {Use Python 2.6} {}
variant python27 conflicts python26 description {Use Python 2.7} {}
if {[variant_isset python26]} {
    python.default_version 26
} else {
    default_variants +python27
    python.default_version 27
}
~~~
Of course you should do this for the suitable versions of python and only if you care about supporting different versions.

**Comment by xeron** [comment]
@mojca thanks for review, I've added different python variants.

**Comment by xeron** [comment]
@mojca could you merge please?

**Comment by g5pw** [comment]
LGTM
